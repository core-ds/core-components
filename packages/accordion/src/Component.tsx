import React, {
    AnchorHTMLAttributes,
    FC,
    KeyboardEvent,
    ReactNode,
    useCallback,
    useState,
} from 'react';
import cn from 'classnames';

import { Typography } from '@alfalab/core-components-typography';

import { DefaultControlIcon } from './components';
import { useRecalculateContentHeight } from './hooks';
import { ControlPosition } from './typings';

import styles from './index.module.css';

export type AccordionProps = {
    /**
     * Состояние компонента
     */
    expanded?: boolean;

    /**
     * Элемент заголовка
     */
    header: ReactNode;

    /**
     * Слот для элемента управления
     */
    control?: ReactNode;

    /**
     * Указывает компоненту - где будет размещен control
     */
    controlPosition?: ControlPosition;

    /**
     * Начальное состояние uncontrolled компонента
     */
    defaultExpanded?: boolean;

    /**
     * Основной элемент для отображения содержимого
     */
    children?: ReactNode;

    /**
     * Дополнительный класс обертки
     */
    className?: string;

    /**
     * Дополнительный класс для контейнера с заголовком
     */
    containerClassName?: string;

    /**
     * Дополнительный класс для header
     */
    headerClassName?: string;

    /**
     * Дополнительный класс для control
     */
    controlClassName?: string;

    /**
     * Дополнительный класс для body
     */
    bodyClassName?: string;

    /**
     * Обработчик смены состояний `expanded`
     */
    onExpandedChange?: (expanded: boolean) => void;

    /**
     * Идентификатор для систем автоматизированного тестирования
     */
    dataTestId?: string;
} & AnchorHTMLAttributes<HTMLDivElement>;

export const Accordion: FC<AccordionProps> = ({
    expanded,
    defaultExpanded = false,
    header,
    control,
    controlPosition = 'end',
    children,
    className,
    containerClassName,
    headerClassName,
    controlClassName,
    bodyClassName,
    onExpandedChange,
    dataTestId,
    ...rest
}) => {
    const uncontrolled = expanded === undefined;
    const [expandedState, setExpanded] = useState(uncontrolled ? defaultExpanded : expanded);
    const isExpanded = uncontrolled ? expandedState : expanded;

    const isStartPosition = controlPosition === 'start';

    const { contentRef, contentCaseRef } = useRecalculateContentHeight(isExpanded);

    const controlContent =
        control === undefined ? (
            <DefaultControlIcon expanded={isExpanded} startPosition={isStartPosition} />
        ) : (
            control
        );

    const headerContent =
        typeof header === 'string' ? (
            <Typography.Text view='primary-large' weight='medium'>
                {header}
            </Typography.Text>
        ) : (
            header
        );

    const bodyContent =
        typeof children === 'string' ? (
            <Typography.Text view='primary-medium'>{children}</Typography.Text>
        ) : (
            children
        );

    const handleExpandedChange = useCallback(() => {
        if (uncontrolled) {
            setExpanded(!isExpanded);
        }

        onExpandedChange?.(!isExpanded);
    }, [isExpanded, onExpandedChange, uncontrolled]);

    const handleKeyDown = (event: KeyboardEvent) => {
        if (event.key === 'Enter') {
            handleExpandedChange();
        }
    };

    return (
        <div {...rest} data-test-id={dataTestId} className={cn(styles.accordion, className)}>
            <div
                role='button'
                tabIndex={0}
                aria-expanded={isExpanded}
                onClick={handleExpandedChange}
                onKeyDown={handleKeyDown}
                className={cn(styles.container, containerClassName)}
            >
                <div
                    className={cn(styles.header, headerClassName, {
                        [styles.endPosition]: isStartPosition,
                    })}
                >
                    {headerContent}
                </div>

                <div
                    className={cn(styles.control, controlClassName, {
                        [styles.startPosition]: isStartPosition,
                    })}
                >
                    {controlContent}
                </div>
            </div>

            <div
                ref={contentRef}
                className={cn(styles.body, bodyClassName, { [styles.expandedBody]: isExpanded })}
            >
                <div className={cn(styles.bodyContent)} ref={contentCaseRef}>
                    {bodyContent}
                </div>
            </div>
        </div>
    );
};
