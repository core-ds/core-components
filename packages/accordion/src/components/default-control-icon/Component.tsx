import React, { FC } from 'react';
import cn from 'classnames';

import { ChevronDownMIcon } from '@alfalab/icons-glyph/ChevronDownMIcon';

import styles from './index.module.css';

type Props = {
    expanded: boolean;
    startPosition: boolean;
};
export const DefaultControlIcon: FC<Props> = ({ expanded, startPosition }) => (
    <ChevronDownMIcon
        className={cn(styles.icon, {
            [styles.expanded]: expanded,
            [styles.startPosition]: startPosition,
        })}
    />
);
