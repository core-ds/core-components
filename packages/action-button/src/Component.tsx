import React, {
    AnchorHTMLAttributes,
    ButtonHTMLAttributes,
    forwardRef,
    Fragment,
    useRef,
} from 'react';
import mergeRefs from 'react-merge-refs';
import cn from 'classnames';

import { Loader } from '@alfalab/core-components-loader';
import { useFocus } from '@alfalab/hooks';

import { useLoader } from './hooks';

import defaultColors from './default.module.css';
import styles from './index.module.css';
import invertedColors from './inverted.module.css';
import staticColors from './static.module.css';

const colorStyles = {
    default: defaultColors,
    inverted: invertedColors,
    static: staticColors,
};

/**
 * Минимальное время отображения лоадера - 500мс,
 * чтобы при быстрых ответах от сервера кнопка не «моргала».
 */
const LOADER_MIN_DISPLAY_INTERVAL = 500;

type Colors = 'default' | 'inverted' | 'static';

type ComponentProps = {
    /**
     * Иконка кнопки
     */
    icon: React.ReactNode;

    /**
     *  Размер кнопки
     */
    size?: 's';

    /**
     * Тип кнопки
     */
    view?: 'primary' | 'secondary';

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Дополнительный класс для обертки иконки
     */
    iconWrapperClassName?: string;

    /**
     * Значение href для ссылки
     */
    href?: string;

    /**
     * Заблокировать кнопку
     */
    disabled?: boolean;

    /**
     * Показать лоадер
     */
    loading?: boolean;

    /**
     * Id компонента для тестов
     */
    dataTestId?: string;

    /**
     * Палитра, в контексте которой используется кнопка
     */
    colors?: Colors;
};

type AnchorProps = ComponentProps & AnchorHTMLAttributes<HTMLAnchorElement>;
type ButtonProps = ComponentProps & ButtonHTMLAttributes<HTMLButtonElement>;
export type ActionButtonProps = Partial<AnchorProps | ButtonProps>;

export const ActionButton = forwardRef<HTMLAnchorElement | HTMLButtonElement, ActionButtonProps>(
    (
        {
            className,
            icon,
            children,
            href,
            size = 's',
            view = 'primary',
            type = 'button',
            iconWrapperClassName,
            disabled,
            loading,
            dataTestId,
            colors = 'default',
            ...rest
        },
        ref,
    ) => {
        const componentRef = useRef<HTMLElement>(null);

        const [focused] = useFocus(componentRef, 'keyboard');

        const { showLoader } = useLoader(!!loading, LOADER_MIN_DISPLAY_INTERVAL);

        const componentProps = {
            className: cn(
                styles.component,
                colorStyles[colors][view],
                styles[size],
                {
                    [styles.focused]: focused,
                    [styles.disabled]: disabled,
                    [styles.loading]: showLoader,
                },
                className,
            ),
            'data-test-id': dataTestId,
        };

        const buttonChildren = (
            <Fragment>
                <span
                    role='img'
                    className={cn(
                        styles.iconWrapper,
                        colorStyles[colors].iconWrapper,
                        styles[size],
                        iconWrapperClassName,
                    )}
                >
                    {showLoader ? <Loader dataTestId='loader' /> : icon}
                </span>
                <span className={styles.label}>{children}</span>
            </Fragment>
        );

        if (href) {
            return (
                <a
                    role='button'
                    ref={mergeRefs([componentRef, ref])}
                    href={href}
                    aria-disabled={disabled || loading}
                    {...componentProps}
                    {...(rest as AnchorHTMLAttributes<HTMLAnchorElement>)}
                >
                    {buttonChildren}
                </a>
            );
        }

        return (
            <button
                ref={mergeRefs([componentRef, ref])}
                // eslint-disable-next-line react/button-has-type
                type={type as ButtonHTMLAttributes<HTMLButtonElement>['type']}
                disabled={disabled || loading}
                {...componentProps}
                {...(rest as ButtonHTMLAttributes<HTMLButtonElement>)}
            >
                {buttonChildren}
            </button>
        );
    },
);
