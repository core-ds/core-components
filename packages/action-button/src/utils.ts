import { getDataTestId } from '@alfalab/core-components-shared';

export function getActionButtonTestIds(dataTestId: string) {
    return {
        actionButton: dataTestId,
        spinner: getDataTestId(dataTestId, 'loader'),
    };
}
