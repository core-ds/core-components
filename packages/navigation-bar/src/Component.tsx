import React, { FC, ReactNode, useEffect, useRef, useState } from 'react';
import cn from 'classnames';

import { getDataTestId } from '@alfalab/core-components-shared';

import styles from './index.module.css';

export interface NavigationBarProps {
    /**
     * Заголовок
     */
    title?: string;

    /**
     * Подзаголовок
     */
    subtitle?: string;

    /**
     * Контент шапки
     */
    children?: ReactNode;

    /**
     * Выравнивание заголовка
     * @default center
     */
    align?: 'left' | 'center';

    /**
     * Цвет фона
     */
    backgroundColor?: string;

    /**
     * Наличие бордера
     */
    border?: boolean;

    /**
     * Фиксирует шапку
     */
    sticky?: boolean;

    /**
     * Слот снизу
     */
    bottomAddons?: ReactNode;

    /**
     * Слот слева
     */
    leftAddons?: ReactNode;

    /**
     * Слот справа
     */
    rightAddons?: ReactNode;

    /**
     * Дополнительный класс для контента
     */
    contentClassName?: string;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Дополнительный класс для правого аддона
     */
    rightAddonsClassName?: string;

    /**
     * Дополнительный класс для левого аддона
     */
    leftAddonsClassName?: string;

    /**
     * Дополнительный класс для нижнего аддона
     */
    bottomAddonsClassName?: string;

    /**
     * Идентификатор для систем автоматизированного тестирования.
     */
    dataTestId?: string;
}

export const NavigationBar: FC<NavigationBarProps> = ({
    align = 'center',
    rightAddons,
    leftAddons,
    bottomAddons,
    sticky,
    border,
    subtitle,
    title,
    children,
    backgroundColor = 'var(--color-light-base-bg-primary)',
    className,
    contentClassName,
    rightAddonsClassName,
    leftAddonsClassName,
    bottomAddonsClassName,
    dataTestId,
}) => {
    const [titleMargin, setTitleMargin] = useState({ left: 0, right: 0 });
    const leftAddonsRef = useRef<HTMLDivElement>(null);
    const rightAddonsRef = useRef<HTMLDivElement>(null);
    const hasLeftAddons = leftAddons && align !== 'left';

    useEffect(() => {
        if (hasLeftAddons) {
            const leftAddonsWidth = leftAddonsRef.current?.offsetWidth || 0;
            const rightAddonsWidth = rightAddonsRef.current?.offsetWidth || 0;

            const marginSize = Math.abs(rightAddonsWidth - leftAddonsWidth);
            const shouldAddLeftMargin = rightAddonsWidth - leftAddonsWidth > 0;

            setTitleMargin((prev) => {
                const newState = shouldAddLeftMargin
                    ? { left: marginSize, right: 0 }
                    : { left: 0, right: marginSize };

                const isStateChanged = prev.left !== newState.left || prev.right !== newState.right;

                return isStateChanged ? newState : prev;
            });
        }
    }, [hasLeftAddons, leftAddons, rightAddons]);

    return (
        <div
            className={cn(
                styles.component,
                {
                    [styles.border]: border,
                    [styles.sticky]: sticky,
                },
                className,
            )}
            style={{
                ...(backgroundColor && { backgroundColor }),
            }}
            data-test-id={dataTestId}
        >
            <div className={cn(styles.mainLine, styles[align], contentClassName)}>
                {hasLeftAddons && (
                    <div className={cn(styles.addons, leftAddonsClassName)} ref={leftAddonsRef}>
                        {leftAddons}
                    </div>
                )}

                {children && <div className={styles.children}>{children}</div>}

                {title && (
                    <div
                        className={cn(styles.content, { [styles[align]]: !hasLeftAddons })}
                        style={{
                            ...(align === 'center'
                                ? {
                                      marginLeft: titleMargin.left,
                                      marginRight: titleMargin.right,
                                  }
                                : null),
                        }}
                    >
                        <div
                            className={cn(styles.title, {
                                [styles[align]]: !hasLeftAddons && !subtitle,
                            })}
                            data-test-id={getDataTestId(dataTestId, 'title')}
                        >
                            {title}
                        </div>

                        {subtitle && (
                            <div
                                className={styles.subtitle}
                                data-test-id={getDataTestId(dataTestId, 'subtitle')}
                            >
                                {subtitle}
                            </div>
                        )}
                    </div>
                )}

                {rightAddons && (
                    <div
                        className={cn(styles.rightAddons, styles.addons, rightAddonsClassName)}
                        ref={rightAddonsRef}
                    >
                        {rightAddons}
                    </div>
                )}
            </div>

            {bottomAddons && <div className={bottomAddonsClassName}>{bottomAddons}</div>}
        </div>
    );
};
