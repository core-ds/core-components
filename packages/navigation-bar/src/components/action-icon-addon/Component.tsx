import React from 'react';
import cn from 'classnames';

import { ButtonMobile } from '@alfalab/core-components-button/mobile';
import { ChevronLeftMIcon } from '@alfalab/icons-glyph/ChevronLeftMIcon';
import { CrossMIcon } from '@alfalab/icons-glyph/CrossMIcon';

import styles from './index.module.css';

export interface ActionIconAddonProps extends React.HTMLAttributes<HTMLButtonElement> {
    /**
     * Тип действия
     */
    action?: 'back' | 'floatingBack' | 'close';

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Обработчик клика
     */
    onClick?: () => void;

    /**
     * Идентификатор для систем автоматизированного тестирования.
     */
    dataTestId?: string;
}

const iconComponents = {
    back: <ChevronLeftMIcon />,
    floatingBack: <ChevronLeftMIcon className={styles.floating} />,
    close: <CrossMIcon className={styles.floating} />,
};

export const ActionIconAddon: React.FC<ActionIconAddonProps> = ({
    onClick,
    className,
    action = 'back',
    dataTestId,
    ...htmlAttributes
}) => (
    <ButtonMobile
        view='text'
        size='s'
        onClick={onClick}
        aria-label={action === 'close' ? 'закрыть' : 'назад'}
        className={cn(styles.component, className)}
        dataTestId={dataTestId}
        {...htmlAttributes}
    >
        <div className={cn(styles.iconWrapper, { [styles[action]]: Boolean(styles[action]) })}>
            {iconComponents[action]}
        </div>
    </ButtonMobile>
);
