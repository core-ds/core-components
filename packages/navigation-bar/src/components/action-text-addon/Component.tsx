import React from 'react';
import cn from 'classnames';

import { ButtonDesktop } from '@alfalab/core-components-button/desktop';
import { Typography } from '@alfalab/core-components-typography';

import styles from './index.module.css';

export interface ActionTextAddonProps extends React.HTMLAttributes<HTMLButtonElement> {
    /**
     * Текст после иконки
     */
    text: string;

    /**
     * Тип действия
     */
    action?: 'primary' | 'secondary';

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Обработчик клика
     */
    onClick?: () => void;

    /**
     * Идентификатор для систем автоматизированного тестирования.
     */
    dataTestId?: string;
}

export const ActionTextAddon: React.FC<ActionTextAddonProps> = ({
    text = 'Назад',
    onClick,
    className,
    action,
    dataTestId,
    ...htmlAttributes
}) => (
    <ButtonDesktop
        view='text'
        size='s'
        onClick={onClick}
        aria-label={text}
        className={cn(styles.component, className)}
        dataTestId={dataTestId}
        {...htmlAttributes}
    >
        <Typography.Text
            className={styles.text}
            view='component'
            weight={action === 'primary' ? 'medium' : 'regular'}
        >
            {text}
        </Typography.Text>
    </ButtonDesktop>
);
