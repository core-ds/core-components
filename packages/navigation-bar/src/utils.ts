import { getDataTestId } from '@alfalab/core-components-shared';

export function getNavigationBarTestIds(dataTestId: string) {
    return {
        navigationBar: dataTestId,
        title: getDataTestId(dataTestId, 'title'),
        subtitle: getDataTestId(dataTestId, 'subtitle'),
    };
}
