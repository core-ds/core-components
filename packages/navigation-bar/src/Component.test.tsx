import React from 'react';
import { render } from '@testing-library/react';
import { NavigationBar, NavigationBarProps } from './Component';
import { getNavigationBarTestIds } from './utils';

const dti = 'nb-dti';

const NavigationBarWrapper = (props: Partial<NavigationBarProps>) => {
    return <NavigationBar dataTestId={dti} title='Title' subtitle='subtitle' {...props} />;
};

describe('NavigationBar', () => {
    describe('Snapshots tests', () => {
        it('should match default snapshot', () => {
            expect(
                render(
                    <NavigationBar
                        dataTestId={dti}
                        title='Title'
                        subtitle='subtitle'
                        leftAddons={<span>Left</span>}
                        rightAddons={<span>Right</span>}
                        bottomAddons={<span>bottom</span>}
                    />,
                ),
            ).toMatchSnapshot();
        });
    });

    describe('Attributes tests', () => {
        it('should set `data-test-id` attribute', () => {
            const { getByTestId } = render(<NavigationBarWrapper />);

            const testIds = getNavigationBarTestIds(dti);

            expect(getByTestId(testIds.navigationBar)).toBeInTheDocument();
            expect(getByTestId(testIds.title)).toBeInTheDocument();
            expect(getByTestId(testIds.subtitle)).toBeInTheDocument();
            expect(getByTestId(dti).tagName).toBe('DIV');
        });
    });

    describe('Classes tests', () => {
        it('should set `className` class', () => {
            const className = 'test-class';
            const { getByTestId } = render(<NavigationBarWrapper className={className} />);

            expect(getByTestId(dti)).toHaveClass(className);
        });

        it('should set custom backgroundColor', () => {
            const backgroundColor = '#fff';
            const { container } = render(
                <NavigationBarWrapper backgroundColor={backgroundColor} />,
            );

            expect(container.firstChild).toHaveStyle('background-color: #fff');
        });

        it('should set `border` class', () => {
            const { getByTestId } = render(<NavigationBarWrapper border={true} />);

            expect(getByTestId(dti)).toHaveClass('border');
        });

        it('should set `sticky` class', () => {
            const { getByTestId } = render(<NavigationBarWrapper sticky={true} />);

            expect(getByTestId(dti)).toHaveClass('sticky');
        });

        it('should set `align` default class', () => {
            const { container } = render(<NavigationBarWrapper />);

            const wrapper = container.firstChild;
            expect(wrapper?.firstChild).toHaveClass('center');
        });

        it('should set `align = left` default class', () => {
            const { container } = render(<NavigationBarWrapper align='left' />);

            const wrapper = container.firstChild;
            expect(wrapper?.firstChild).toHaveClass('left');
        });
    });

    it('should unmount without errors', () => {
        const { unmount } = render(<NavigationBarWrapper />);

        expect(unmount).not.toThrowError();
    });
});
