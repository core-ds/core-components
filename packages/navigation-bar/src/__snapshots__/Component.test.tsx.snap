// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Snapshots tests should match default snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="component"
        data-test-id="nb-dti"
      >
        <div
          class="mainLine center"
        >
          <div
            class="addons"
          >
            <span>
              Left
            </span>
          </div>
          <div
            class="content"
            style="margin-left: 0px; margin-right: 0px;"
          >
            <div
              class="title"
              data-test-id="nb-dti-title"
            >
              Title
            </div>
            <div
              class="subtitle"
              data-test-id="nb-dti-subtitle"
            >
              subtitle
            </div>
          </div>
          <div
            class="rightAddons addons"
          >
            <span>
              Right
            </span>
          </div>
        </div>
        <div>
          <span>
            bottom
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="component"
      data-test-id="nb-dti"
    >
      <div
        class="mainLine center"
      >
        <div
          class="addons"
        >
          <span>
            Left
          </span>
        </div>
        <div
          class="content"
          style="margin-left: 0px; margin-right: 0px;"
        >
          <div
            class="title"
            data-test-id="nb-dti-title"
          >
            Title
          </div>
          <div
            class="subtitle"
            data-test-id="nb-dti-subtitle"
          >
            subtitle
          </div>
        </div>
        <div
          class="rightAddons addons"
        >
          <span>
            Right
          </span>
        </div>
      </div>
      <div>
        <span>
          bottom
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
